From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrei Aldea <andrei1998@gmail.com>
Date: Tue, 4 Aug 2026 10:49:43 -0500
Subject: [PATCH] arm64: dts: ti: add BeagleBadge GDEY042T81 overlay

Add a device-tree overlay for the BeagleBadge SPI-connected Good Display
GDEY042T81 e-paper panel and include it in the TI K3 DTB build.

Developed with assistance from OpenAI Codex.

Signed-off-by: Andrei Aldea <andrei1998@gmail.com>
---
 arch/arm64/boot/dts/ti/Makefile               |  1 +
 .../ti/k3-am62l3-badge-eink-gdey042t81.dtso   | 62 +++++++++++++++++++
 2 files changed, 63 insertions(+)
 create mode 100644 arch/arm64/boot/dts/ti/k3-am62l3-badge-eink-gdey042t81.dtso

diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
index cb2623885b0c..6c5d6dd4b3a6 100644
--- a/arch/arm64/boot/dts/ti/Makefile
+++ b/arch/arm64/boot/dts/ti/Makefile
@@ -62,4 +62,5 @@
 dtb-$(CONFIG_ARCH_K3) += k3-am62l3-evm-epaper.dtbo
 dtb-$(CONFIG_ARCH_K3) += k3-am62l3-evm-eqep.dtbo
+dtb-$(CONFIG_ARCH_K3) += k3-am62l3-badge-eink-gdey042t81.dtbo
 dtb-$(CONFIG_ARCH_K3) += k3-am62l3-evm-mcan.dtbo
 dtb-$(CONFIG_ARCH_K3) += k3-am62l3-evm-pwm.dtbo
diff --git a/arch/arm64/boot/dts/ti/k3-am62l3-badge-eink-gdey042t81.dtso b/arch/arm64/boot/dts/ti/k3-am62l3-badge-eink-gdey042t81.dtso
new file mode 100644
index 000000000000..3295568b416f
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-am62l3-badge-eink-gdey042t81.dtso
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/**
+ * DT Overlay for GoodDisplay GDEY042T81 E-Ink Display on AM62L Badge
+ *
+ * Copyright (C) 2025 Texas Instruments Incorporated - https://www.ti.com/
+ */
+
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include "k3-pinctrl.h"
+
+&pmx0 {
+
+	dc_pins_default: dc-default-pins {
+		pinctrl-single,pins = <
+			AM62LX_IOPAD(0x01ac, PIN_OUTPUT, 7) /* (E12) SPI0_D0.GPIO0_90 */
+		>;
+	};
+
+	res_pins_default: res-default-pins {
+		pinctrl-single,pins = <
+			AM62LX_IOPAD(0x009c, PIN_OUTPUT, 7) /* (H20) GPMC0_AD9.GPIO0_24 */
+		>;
+	};
+
+	busy_pins_default: busy-default-pins {
+		pinctrl-single,pins = <
+			AM62LX_IOPAD(0x00dc, PIN_INPUT, 7) /* (N21) GPMC0_WPn.GPIO0_39 */
+		>;
+	};
+
+	main_spi0_pins_default: main-spi0-default-pins {
+		pinctrl-single,pins = <
+			AM62LX_IOPAD(0x01a8, PIN_OUTPUT, 0) /* (E13) SPI0_CLK */
+			AM62LX_IOPAD(0x01a0, PIN_OUTPUT, 0) /* (E11) SPI0_CS0 */
+			AM62LX_IOPAD(0x01b0, PIN_OUTPUT, 0) /* (B12) SPI0_D1 */
+		>;
+	};
+};
+
+&spi0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&main_spi0_pins_default>;
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	display@0 {
+		compatible = "gooddisplay,gdey042t81";
+		reg = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&dc_pins_default>,
+			    <&res_pins_default>,
+			    <&busy_pins_default>;
+		spi-max-frequency = <2000000>;
+		reset-gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+		dc-gpios = <&gpio0 90 GPIO_ACTIVE_HIGH>;
+		busy-gpios = <&gpio0 39 GPIO_ACTIVE_HIGH>;
+	};
+};
